doc/source/configuration/ci-cd.rst

```diff
@@ -0,0 +1,179 @@
+
+Runner Deployment
+-----------------
+
+1. Identify a suitable host for hosting the runners.
+    GitHub runners need to be deployed on a host which has not had Docker deployed using kolla.
+    This is because GitHub runners cannot provide `network options when running in a container <https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#jobsjob_idcontaineroptions>`__.
+
+    Ideally an Infra VM could be used here or failing that the control host.
+    Wherever it is deployed the host will need access to the :code:`admin_network`, :code:`public_network` and the :code:`pulp registry` on the seed.
+
+2. Edit the environment's :code:`${KAYOBE_CONFIG_PATH}/environments/${KAYOBE_ENVIRONMENT}/inventory/groups` to add the predefined :code:`github-runners` group to :code:`infra-vms`
+
+.. code-block:: ini
+
+    [infra-vms:children]
+    github-runners
+
+3. Edit the environment's :code:`${KAYOBE_CONFIG_PATH}/environments/${KAYOBE_ENVIRONMENT}/inventory/hosts` to define the host(s) that will host the runners.
+
+.. code-block:: ini
+
+    [github-runners]
+    runner-01
+
+4. Provide all the relevant Kayobe :code:`group_vars` for :code:`github-runners` under :code:`${KAYOBE_CONFIG_PATH}/environments/${KAYOBE_ENVIRONMENT}/inventory/group_vars/github-runners`
+    * `infra-vms` ensuring all required `infra_vm_extra_network_interfaces` are defined
+    * `network-interfaces`
+    * `python-interpreter.yml` ensuring that `ansible_python_interpreter: /usr/bin/python3` has been set
+
+5. Create or edit `${KAYOBE_CONFIG_PATH}/inventory/group_vars/github-runners/runners.yml` file which will contain the variables required to deploy a series of runners
+
+.. code-block:: yaml
+
+    ---
+    runner_user: VM_USER_NAME_HERE
+    github_account: ORG_NAME_HERE
+    github_repo: KAYOBE_CONFIG_REPO_NAME_HERE
+    access_token: "{{ secrets_github_access_token }}"
+
+    base_runner_dir: /opt/actions-runner
+
+    default_runner_labels:
+      - kayobe
+      - openstack
+      - "{{ kayobe_environment | default(omit) }}"
+
+    # Dictionary of runners to be deployed using the role.
+    # Each dict item can be provided with optional attributes
+    # * labels - provide a list of labels for a specific runner
+    #   overriding the contents of `default_runner_labels`
+    # * state - either `started`` or `absent`. By default it will
+    #   be started if however the runner needs to be removed
+    #   then setting it to `absent` will unregister the runner with
+    #   GitHub and remove it from the system.
+    # Example
+    # github_runners:
+    #  runner_01: {}
+    #  runner_02:
+    #    labels: ['foo', 'bar', 'baz']
+    #  runner_03:
+    #    state: absent
+    github_runners:
+      runner_01: {}
+      runner_02: {}
+      runner_03: {}
+
+    docker_users:
+      - "{{ runner_user }}"
+
+    pip_install_packages:
+      - name: docker
```

Comment:
We don't need to include the whole variables file now that we have it in the config - it will only get out of sync. You could list the mandatory variables, or provide an example of a config including just those.

Reply:
Removed the static variables and focused on role/API specific variables